internal/collector/processor/processor.go

```diff
@@ -37,11 +43,109 @@ func (sp *debugSpanProcessor) ProcessSpans(batch *agenttracepb.ExportTraceServic
+
+	sp.logger.Debug("debugSpanProcessor", zap.String("originalFormat", spanFormat), zap.Int("#spans", numSpans))
 	return 0, nil
 }
 
 // NewNoopSpanProcessor creates an OC SpanProcessor that just drops the received data.
 func NewNoopSpanProcessor(logger *zap.Logger) SpanProcessor {
 	return &debugSpanProcessor{logger: logger}
 }
+
+// Keys and stats for telemetry.
+var (
+	tagSourceFormatKey, _ = tag.NewKey("format")
+	tagServiceNameKey, _  = tag.NewKey("service")
+	tagExporterNameKey, _ = tag.NewKey("exporter")
+
+	statBatchCount        = stats.Int64("batches_received", "counts the number of batches received", stats.UnitDimensionless)
+	statDroppedBatchCount = stats.Int64("batches_dropped", "counts the number of batches dropped", stats.UnitDimensionless)
+
+	statSpanCount        = stats.Int64("spans_received", "counts the number of spans received", stats.UnitDimensionless)
+	statDroppedSpanCount = stats.Int64("spans_dropped", "counts the number of spans dropped", stats.UnitDimensionless)
+)
+
+// metricTagKeys returns the metric tag keys according to the given telemetry level.
+func metricTagKeys(level telemetry.Level) []tag.Key {
+	var tagKeys []tag.Key
+	switch level {
+	case telemetry.Detailed:
+		tagKeys = append(tagKeys, tagServiceNameKey)
+		fallthrough
+	case telemetry.Normal:
+		tagKeys = append(tagKeys, tagSourceFormatKey)
+		fallthrough
+	case telemetry.Basic:
+		tagKeys = append(tagKeys, tagExporterNameKey)
+		break
+	default:
+		return nil
+	}
+
+	return tagKeys
+}
+
+// MetricViews return the metrics views according to given telemetry level.
+func MetricViews(level telemetry.Level) []*view.View {
+	tagKeys := metricTagKeys(level)
+	if tagKeys == nil {
+		return nil
+	}
+
+	// There are some metrics enabled, return the views.
+	receivedBatchesView := &view.View{
+		Name:        statBatchCount.Name(),
+		Measure:     statBatchCount,
+		Description: "The number of span batches received.",
+		TagKeys:     tagKeys,
+		Aggregation: view.Count(),
+	}
+	droppedBatchesView := &view.View{
+		Name:        statDroppedBatchCount.Name(),
+		Measure:     statDroppedBatchCount,
+		Description: "The number of span batches dropped.",
+		TagKeys:     tagKeys,
+		Aggregation: view.Count(),
+	}
+	receivedSpansView := &view.View{
+		Name:        statSpanCount.Name(),
+		Measure:     statSpanCount,
+		Description: "The number of spans received.",
+		TagKeys:     tagKeys,
+		Aggregation: view.Sum(),
+	}
+	droppedSpansView := &view.View{
+		Name:        statDroppedSpanCount.Name(),
+		Measure:     statDroppedSpanCount,
+		Description: "The number of spans dropped.",
+		TagKeys:     tagKeys,
+		Aggregation: view.Sum(),
+	}
+
+	return []*view.View{receivedBatchesView, droppedBatchesView, receivedSpansView, droppedSpansView}
+}
+
+func statsTagsForBatch(processorName string, batch *agenttracepb.ExportTraceServiceRequest, spanFormat string) []tag.Mutator {
+	var serviceName string
```

Comment:
really nitpicky comment, but I would suggest to break out the logic below into a separate utility function "serviceNameForBatch" and pass in a serviceName param to this method similar to processorName instead of the batch itself

Reply:
Good idea, I like it :)